tests/storage/blockvolume_test.py

```diff
@@ -144,9 +144,12 @@ def test_optimal_size_cow_leaf(self):
         with self.make_volume(size=2 * GiB, format=sc.COW_FORMAT) as vol:
             chunk_size = GiB
             check = qemuimg.check(vol.getVolumePath(), qemuimg.FORMAT.QCOW2)
-            actual_size = utils.round(
+            optimal_size = utils.round(
                 check['offset'] + chunk_size, vol.align_size)
-            self.assertEqual(vol.optimal_size(), actual_size)
+            self.assertEqual(vol.optimal_size(), optimal_size)
+            self.assertEqual(
+                vol.optimal_cow_size(check["offset"], 2 * GiB, True),
```

Comment:
Something like this would be clearer?
```
vol.optimal_cow_size(check["offset"], capacity=2 * GiB, is_lead=True),
```
At least having a name for the boolean literal parameter at the end would help, so that we don't have to scroll up to the definition to see what it means in this position.

Reply:
Yes, is_leaf= is certainly needed here.

Reply:
While is_leaf=True is more clear, it treats the argument as keyword argument, and we try to always call positional arguments as:

    func(arg1, arg2)

and keyword arguments as:

    func(key=value)

This way code never breaks when keyword argument order is changed (e.g. adding new keyword argument), or when renaming positional argument. Both changes are valid when callers use these rules.

So base on our current conventions, if we want to use is_leaf=True, we need to add default value, which make it more fragile - easy to call with the missing argument and get the wrong result.

Reply:
Improved by using `vol.isLeaf()` instead of True which is also more similar to the actual code.